### PR TITLE
Time Tracking Settings Methods

### PR DESCRIPTION
**Pull request by aschoelzhorn**
Implemented missing [Time Tracking Settings Methods](https://www.jetbrains.com/help/youtrack/standalone/Time-Tracking-Settings-Methods.html):

* [GET System-wide Time Tracking Settings](https://www.jetbrains.com/help/youtrack/standalone/GET-System-wide-Time-Tracking-Settings.html)
* [PUT Time Tracking Settings for a Project](https://www.jetbrains.com/help/youtrack/standalone/PUT-Time-Tracking-Settings-for-a-Project.html)
* [GET Time Tracking Settings for a Project](https://www.jetbrains.com/help/youtrack/standalone/GET-Time-Tracking-Settings-for-a-Project.html)
* [PUT System-wide Time Tracking Settings](https://www.jetbrains.com/help/youtrack/standalone/PUT-System-wide-Time-Tracking-Settings.html)

### CODE REVIEW

File: src/YouTrackSharp/TimeTracking/TimeSettings.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		public int HoursADay { get; set; }
+
+		/// <summary>
+		/// Days A Week
+		/// </summary>
+		[JsonProperty("daysAWeek")]
+		public int DaysAWeek { get; set; }
+
+		[JsonProperty("workWeek")]
+		public List<WorkDay> Workweek { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/TimeTracking/TimeSettings.cs]: Can you add XMLdoc here?

File: src/YouTrackSharp/TimeTracking/TimeSettings.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+		public class WorkDay
```
Comment (maartenba) [inline src/YouTrackSharp/TimeTracking/TimeSettings.cs]: Chack if this class can be removed and `SubValue<int>` can be used instead?
(https://github.com/JetBrains/YouTrackSharp/blob/master/src/YouTrackSharp/Entities/SubValue.cs)

File: src/YouTrackSharp/TimeTracking/TimeSettings.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace YouTrackSharp.TimeTracking
+{
+	/// <summary>
+	/// A class that represents YouTrack time settings information.
+	/// </summary>
+	public class TimeSettings
```
Comment (maartenba) [inline src/YouTrackSharp/TimeTracking/TimeSettings.cs]: Should we consider renaming to `TimeTrackingSettings`? (to stay close to the REST API description)
I see this is used for the "global" settings so perhaps `SystemWideTimeTrackingSettings` ?

File: src/YouTrackSharp/TimeTracking/TimeTrackingService.cs [hunk 4]
```diff
@@ -196,5 +196,130 @@ public TimeTrackingService(Connection connection)
...
-    }
+
+
+		/// <summary>
+		/// Get the current time tracking settings for a specific project.
+		/// </summary>
+		/// <remarks>Uses the REST API <a href="https://www.jetbrains.com/help/youtrack/standalone/GET-System-wide-Time-Tracking-Settings.html">Get System-wide Time Tracking Settings</a>.</remarks>
+		/// <returns>System-wide <see cref="TimeSettings" />.</returns>
+		/// <exception cref="T:System.Net.HttpRequestException">When the call to the remote YouTrack server instance failed.</exception>
+		public async Task<TimeSettings> GetSystemwideTimeTrackingSettings()
```
Comment (maartenba) [inline src/YouTrackSharp/TimeTracking/TimeTrackingService.cs]: `GetSystemwideTimeTrackingSettings` -> `GetSystem**W**ideTimeTrackingSettings`

File: src/YouTrackSharp/TimeTracking/TimeTrackingService.cs [hunk 5]
```diff
@@ -196,5 +196,130 @@ public TimeTrackingService(Connection connection)
...
+
+
+		/// <summary>
+		/// Updates the current time tracking settings for a specific project.
+		/// </summary>
+		/// <remarks>Uses the REST API <a href="https://www.jetbrains.com/help/youtrack/standalone/PUT-Time-Tracking-Settings-for-a-Project.html">Set system-wide time tracking settings: a list of working days in a week, and a number of hours in a working day</a>.</remarks>
+		/// <param name="timeSettings"><see cref="TimeSettings" />Parameter daysAWeek is ignored since Youtrack 5.1</param>
+		/// <exception cref="T:YouTrackErrorException">When the call to the remote YouTrack server instance failed and YouTrack reported an error message.</exception>
+		/// <exception cref="T:System.Net.HttpRequestException">When the call to the remote YouTrack server instance failed.</exception>
+		public async Task UpdateSystemwideTimeTrackingSettings(TimeSettings timeSettings)
```
Comment (maartenba) [inline src/YouTrackSharp/TimeTracking/TimeTrackingService.cs]: `UpdateSystemwideTimeTrackingSettings` -> `UpdateSystem**W**ideTimeTrackingSettings`

**Comment by aschoelzhorn** [comment]
I have made the proposed changes and fixed also some comments

**Comment by maartenba** [comment]
Looking good, thanks! Will do a bit more work on testing and then create a 3.1.0-beta-2 release.

**Pull request merged**